tests/conftest.py

```diff
@@ -166,33 +152,56 @@ def init_microvm(root_path, bin_cloner_path, features=''):
     return vm
 
 
-@pytest.fixture(autouse=True, scope='session')
-def test_session_root_path():
-    """Ensure and yield the testrun root directory.
+def pytest_configure(config):
+    """Pytest hook - initialization.
+
+    Initialize the test scheduler and IPC services.
+    """
+    PytestScheduler.instance().register_mpsing(
+        net_tools.UniqueIPv4Generator.instance()
+    )
+    config.pluginmanager.register(PytestScheduler.instance())
+
 
-    If created here, it is also destroyed during teardown. The root directory
-    is created per test session.
+def pytest_addoption(parser):
+    """Pytest hook. Add concurrency command line option.
+
+    For some reason, pytest doesn't properly pick up this hook in our plugin
+    class, so we need to call it from here.
     """
-    created_test_session_root_path = False
+    return PytestScheduler.instance().do_pytest_addoption(parser)
 
-    try:
-        root_path = os.environ[ENV_TMPDIR_VAR]
-    except KeyError:
-        root_path = DEFAULT_ROOT_TESTSESSION_PATH
 
-    try:
-        delete_test_session = (len(os.environ[ENV_KEEP_TEST_SESSION]) == 0)
-    except KeyError:
-        delete_test_session = True
+@pytest.fixture(autouse=True, scope='session')
+def test_session_root_path():
+    """Ensure and yield the testrun root directory.
 
-    if not os.path.exists(root_path):
-        os.makedirs(root_path)
-        created_test_session_root_path = True
+    Created at session initialization time, this directory will be
```

Comment:
This looks to be equivalent to mktemp -d. In python is already implemend by tempfile.